component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java

```diff
@@ -0,0 +1,295 @@
+                            attributeExpressionExecutors[2].getReturnType());
+                }
+            } else {
+                throw new SiddhiAppCreationException("decayRate has to be a constant but found " +
+                        this.attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+        } else {
+            throw new SiddhiAppCreationException("The second query parameter should either be decayRate or " +
+                    "numberOfClusters which should be of type double or int respectively but found " +
+                    attributeExpressionExecutors[1].getReturnType());
+        }
+
+        dimensionality = attributeExpressionExecutors.length - coordinateStartIndex;
+        coordinateValuesOfCurrentDataPoint = new double[dimensionality];
+
+        //validating all the attributes to be variables
+        for (int i = coordinateStartIndex; i < coordinateStartIndex + dimensionality; i++) {
+            if (!(this.attributeExpressionExecutors[i] instanceof VariableExpressionExecutor)) {
+                throw new SiddhiAppCreationException("The attributes should be variable but found a " +
+                        this.attributeExpressionExecutors[i].getClass().getCanonicalName());
+            }
+        }
+
+        String siddhiAppName = siddhiAppContext.getName();
+        modelName = modelName + "." + siddhiAppName;
+        if (logger.isDebugEnabled()) {
+            logger.debug("model name is " + modelName);
+        }
+
+        clusterer = new Clusterer(numberOfClusters, 2, modelName, siddhiAppName, dimensionality);
+
+        List<Attribute> attributeList = new ArrayList<>(1 + dimensionality);
+        attributeList.add(new Attribute("euclideanDistanceToClosestCentroid", Attribute.Type.DOUBLE));
+        for (int i = 1; i <= dimensionality; i++) {
+            attributeList.add(new Attribute("closestCentroidCoordinate" + i, Attribute.Type.DOUBLE));
+        }
+        return attributeList;
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor processor,
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+        synchronized (this) {
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+
+                //validating and getting coordinate values
+                for (int i = coordinateStartIndex; i < coordinateStartIndex + dimensionality; i++) {
+                    try {
+                        Number content = (Number) attributeExpressionExecutors[i].execute(streamEvent);
+                        coordinateValuesOfCurrentDataPoint[i - coordinateStartIndex] = content.doubleValue();
+                    } catch (ClassCastException e) {
+                        throw new SiddhiAppCreationException("coordinate values should be int/float/double/long " +
```

Comment:
@niruhan this should be SiddhiAppRuntimeException

Reply:
Fixed in https://github.com/siddhi-io/siddhi-execution-streamingml/pull/76